chain/network/src/peer/peer_actor.rs

```diff
@@ -159,13 +159,31 @@ struct HandshakeSpec {
     partial_edge_info: PartialEdgeInfo,
 }
 
+type HandshakeSignalSender = tokio::sync::oneshot::Sender<std::convert::Infallible>;
+pub type HandshakeSignal = tokio::sync::oneshot::Receiver<std::convert::Infallible>;
+
 impl PeerActor {
-    pub(crate) fn spawn(
+    /// Spawns a PeerActor on a separate actix::Arbiter and awaits for the
+    /// handshake to succeed/fail. The actual result is not returned because
+    /// actix makes everything complicated.
+    pub(crate) async fn spawn_and_handshake(
         clock: time::Clock,
         stream: tcp::Stream,
         force_encoding: Option<Encoding>,
         network_state: Arc<NetworkState>,
     ) -> anyhow::Result<actix::Addr<Self>> {
+        let (addr, handshake_signal) = Self::spawn(clock, stream, force_encoding, network_state)?;
+        // This is a receiver of Infallible, so it only completes when the channel is closed.
```

Comment:
can you explain more?

Reply:
done

Reply:
I was more wondering why we have .err().unwrap() there.

as normally I'd have expected the handshake signal to return OK if handshake was successful and Error otherwise.